uqcsbot/scripts/xkcd.py

```diff
@@ -0,0 +1,75 @@
+import requests
+import feedparser
+from urllib.parse import quote
+from uqcsbot import bot, Command
+from uqcsbot.utils.command_utils import loading_status
+
+
+# HTTP Endpoints
+XKCD_BASE_URL = "https://xkcd.com/"
+XKCD_RSS_URL = "https://xkcd.com/rss.xml"
+RELEVANT_XKCD_URL = 'https://relevantxkcd.appspot.com/process?action=xkcd&query='
+
+
+def get_by_id(comic_number: int) -> str:
+    """
+    Gets an xkcd comic based on its unique ID/sequence number.
+    :param comic_number: the ID number of the xkcd comic to retrieve.
+    :return: a response containing either a comic URL or an error message.
+    """
+    if comic_number < 0:
+        return "Invalid xkcd ID, it must be a positive integer."
+    url = f"{XKCD_BASE_URL}{str(comic_number)}"
+    response = requests.get(url)
+    if response.status_code == 200:
+        return url
+    else:
+        return "Could not retrieve an xkcd with that ID (are there even that many?)"
+
+
+def get_by_search_phrase(phrase: str) -> str:
+    """
+    Uses the site relevantxkcd.appspot.com to identify the most appropriate xkcd comic
+    based on the phrase provided.
+    :param phrase: the phrase to find an xkcd comic related to.
+    :return: the URL of the most relevant comic for that search phrase.
+    """
+    url = f"{RELEVANT_XKCD_URL}{quote(phrase)}"
+    response = requests.get(url)
+    relevant_comics = response.content.decode().split("\n")[2:]
```

Comment:
nit: maybe provide a comment on what's going on over the next few lines-- maybe a snippet example?

Reply:
I think I've addressed this, let me know if what I added is all g.